table-structure.sql

```diff
@@ -0,0 +1,59 @@
+--User: id (pk), username, email, password
+--Role: id (pk), name
+--UserRoles: id(pk), user_id (fk), role_id (fk)   //A user may have multiple roles [Unique constraint on user_id, role_id]
+--Convention: id (pk), name, year   //startDate?, endDate?
+--Package: id (pk), name     //More fields for the "package" bought? ie include_shirt, num_of_days, etc?
+--Attendee: id (pk), user_id (fk), convention_id (fk), package_id (fk) [unique constraint on user_id, convention_id, package_id]
+
+
+
+CREATE TABLE user (
```

Comment:
We might want to put the user's first and last name here as nullable fields. We need that info to build the name badges later. I could see a case for having the name associated with the individual `attendee`s below if a single user could buy con tickets for multiple people.

Reply:
Erp, as well as location and preferred ME race, if we are going to have all of these.

Having these fields as well really makes me think they belong someone outside of the `user` proper, which feels like a authentication entity. Thoughts?

Reply:
Yeah, I debated putting first name / last name in this table. I do like the thought of another table with user profile information - name, location, ME race, anything else we want

```
user_info (
id
user_id - from user
first_name
last_name
online_handle
me_location
me_race
)
```